app/models/course.rb

```diff
@@ -117,6 +119,31 @@ def root_folder?
     end
   end
 
+  def initialize_duplicate(duplicator, other)
+    self.start_at += duplicator.time_shift
+    self.end_at += duplicator.time_shift
+    self.title = duplicator.new_course_title
+
+    # This also duplicates assessments.
+    self.lesson_plan_items = duplicator.duplicate(other.lesson_plan_items.map(&:actable)).
```

Comment:
Didn't work. The actable gem can't convert the associations to the correct sequence of database records when done that way.